bspump/declarative/expression/statement/whenexpr.py

```diff
@@ -24,29 +26,86 @@ class WHEN(Expression):
 
 	"""
 
-	Attributes = {
-		"Items": [],
-	}
+	Attributes = False  # Filled during initialize() since attributes are dynamic
+
 
 	def __init__(self, app, *, sequence):
 		super().__init__(app)
 		self.Items = sequence
+		self.ItemsNormalized = []
+		self.Attributes = {}
+		self.OutletType = None  # Will be determined in `initialize()`
+		self.Else = VALUE(self.App, value=None)
 
-	def __call__(self, context, event, *args, **kwargs):
-		for branch in self.Items:
-			expr_test = branch.get('test')
-			if expr_test is not None:
+	def set(self, key, value):
```

Comment:
I don't get this one ... when this is called?
It seems to me to be unnecessary, Expressions are quite immutable objects - they don't change after initialisation.
So why this is needed?

Reply:
@ateska Set is called during optimization process to replace the unoptimzed expression with the optimized one. However, this applies only to expression attributes such as self.Then0, but in self.ItemsOptimized the expression stays the same (unoptimized). Therefore, when self.Then0 is changed, the corresponding item must also be changed in self.ItemsOptimized.